### PR TITLE
Improve BwC for returning clause for update

### PR DESCRIPTION
**Pull request by mkleen**
## Summary of the changes / Why this improves CrateDB

This introduces a version check for the returning clause for update in the planner. It also removes all `allOn4_2` version flags from upsert related classes and uses the version from the stream directly as @seut suggested https://github.com/crate/crate/pull/9434#discussion_r371270034. This pr will also  fixes the flaky RollingUpgradeTest in crate-qa against current master.

Related BwC test can be found here https://github.com/crate/crate-qa/pull/131/

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/main/java/io/crate/planner/consumer/UpdatePlanner.java [hunk 1]
```diff
@@ -82,14 +83,25 @@
...
     private UpdatePlanner() {
     }
 
     public static Plan plan(AnalyzedUpdateStatement update,
                             Functions functions,
                             PlannerContext plannerCtx,
                             SubqueryPlanner subqueryPlanner) {
+
+        if (!update.returnValues().isEmpty()) {
+            if (!plannerCtx.clusterState().getNodes().getMinNodeVersion().onOrAfter(Version.V_4_2_0)) {
```
Comment (mkleen) [inline sql/src/main/java/io/crate/planner/consumer/UpdatePlanner.java]: Which version are we aiming here 4.2 ?

Comment (mfussenegger) [reply sql/src/main/java/io/crate/planner/consumer/UpdatePlanner.java]: 4.2 :+1:

File: sql/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java [hunk 2]
```diff
@@ -225,6 +213,8 @@ public void writeTo(StreamOutput out) throws IOException {
 
         sessionSettings.writeTo(out);
 
+        boolean allOn4_2 = out.getVersion().onOrAfter(Version.V_4_2_0);
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java]: Maybe also inline this, I think there is only a single use.

File: sql/src/main/java/io/crate/planner/consumer/UpdatePlanner.java (hunk 1 above)
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/UpdatePlanner.java]: Could merge this into the previous if condition.

File: sql/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java (hunk 2 above)
Comment (mkleen) [reply sql/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java]: It is used for the serialization of all the items.

Comment (mfussenegger) [reply sql/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java]: Ah right, I missed that.

**Pull request merged**